### PR TITLE
User can view their own profile

### PR DESCRIPTION
**Pull request by KyunginNa**
[User will have a screen to view his/her own posts](https://www.pivotaltracker.com/story/show/176613176)
### User Story
```
As an API,            
In order to provide a client with a personal profile,         
I want to send data about the current user.
```
## Changes proposed in this pull request:
* Writing a test to check if a user can view a profile
* Generating Users controller with show action
* Generating User Show Serializer
* Adding a GET route for users

### CODE REVIEW

File: app/serializers/user_show_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class UserShowSerializer < ActiveModel::Serializer
+  attributes :id, :email
+
```
Comment (salindholm) [inline app/serializers/user_show_serializer.rb]: Please remove this extra line

File: app/controllers/api/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Api::UsersController < ApplicationController
+  def show
+    user = User.find(params['id'])
+    render json: user, serializer: UserShowSerializer
+  rescue StandardError => e
+    render json: { message: e.message }, status: 404
```
Comment (salindholm) [inline app/controllers/api/users_controller.rb:6]: What is the error message?

**Review by salindholm — CHANGES_REQUESTED** [review]
Great job as always, left comments!

File: app/controllers/api/users_controller.rb (hunk 2 above)
Comment (KyunginNa) [reply app/controllers/api/users_controller.rb:6]: The error message is a dynamic one we get from rails! When we send in an invalid id, the message would be "Couldn't find User with 'id'=abc" :)

File: spec/requests/api/user_can_view_profile_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    it 'is expected to return a user email' do
+      expect(response_json['user']['email']).to eq 'test-2@example.com'
+    end
+
+    it 'is expected to return the track of the post' do
+      expect(response_json['user']['posts'][0]['track']).to eq 'MyTrack'
+    end
+
+    it 'is expected to return the descrption of the post' do
```
Comment (rithreaksa) [inline spec/requests/api/user_can_view_profile_spec.rb]: You have a small typo here 😄. You forgot `i` in description

File: app/serializers/user_show_serializer.rb (hunk 1 above)
Comment (KyunginNa) [reply app/serializers/user_show_serializer.rb]: Fixed now!

File: spec/requests/api/user_can_view_profile_spec.rb (hunk 3 above)
Comment (KyunginNa) [reply spec/requests/api/user_can_view_profile_spec.rb]: Fixed now 👍

**Pull request merged**